# ***Calculate Gradient for Mathematical Quadratic Equation***

This code is all about how we can solve an quadratic mathematical equation and compute its derivative using Pytorch. This can compute with Pytorch autograd package which provides differentiation function for a tensor object. 

A tensor object requires an attribute naming as .requires_grad attribute. By providing its value True, it can track all operations that are performed on a Tensor object. Then I have used .backward function for compute the gradients for the equation. By using .grad attribute we can see the gradient value.

In [19]:
#Attach Libraries
import torch

In [20]:
#Create Tensor variable with value and requires_grad = True
x = torch.tensor(5.0,requires_grad=True)

In [21]:
#Mathematical Quadratic Equation
y = 5*x**2+9*x+3

In [22]:
#Calculate the value of y with having value of x, i.e. 5
print(y)

tensor(173., grad_fn=<AddBackward0>)


Answer you get by run the above command

tensor(173., grad_fn=<AddBackward0>)

In [23]:
#Calculate Gradient
#dy/dx = 10*x+9
y.backward()

In [26]:
#Show Gradient Value
x.grad

tensor(59.)

Answer you get by run the above command

tensor(59.)